### ISSUE TITLE
about temperature display problem

### ISSUE THREAD

**Issue by b4ll5oh4rd** [shayne/hwinfo-streamdeck#53]
[](url)
![a](https://user-images.githubusercontent.com/47036391/152313241-380b96b3-de6a-4468-ab52-39c1e9872eb6.jpg)
sensor temperature c° can't display.

HWiNFO64  7.16-4650
STREAM DECK  5.2.0.14948
HWINFO-STREAMDECK 2.0.3
OS Win10 x64

**Comment by b4ll5oh4rd** [comment]
yes ,my win10 region format is English(united states),but system locale is Chinese(simplified,china). i choose use Unicode Utf-8 for worldwide language support, Also does not display properly.I haven't changed the system font.
thank u.
![1](https://user-images.githubusercontent.com/47036391/152461360-e47afaea-b1f4-45ce-a57c-0bcfb122adbc.png)
![2](https://user-images.githubusercontent.com/47036391/152461369-a42f92e4-45cc-44b4-ad87-171488363a29.png)

**Comment by b4ll5oh4rd** [comment]
If I change the system locale to English, the temperature can be displayed normally。
![3](https://user-images.githubusercontent.com/47036391/152462205-02af3c17-a444-4b37-b45c-5949568136df.png)

**Comment by Paly-thoa** [comment]
Hello, As long as you set HWInfo to Fahrenheit or Celcius, then you can use the formatter to display the number correctly.

This will show the number as ###°F
%3.0f°F

This will show the number as ###°C
%3.0f°C

![image](https://user-images.githubusercontent.com/99476709/153573840-17938645-e9a7-4f49-b8e3-772d1b33fd14.png)

**Comment by b4ll5oh4rd** [comment]
[](url)> Hello, As long as you set HWInfo to Fahrenheit or Celcius, then you can use the formatter to display the number correctly. Then you can probably set you locale to what ever you want.
> 
> This will show the number as ###°F %3.0f°F
> 
> This will show the number as ###°C %3.0f°C
> 
> ![image](https://user-images.githubusercontent.com/99476709/153573840-17938645-e9a7-4f49-b8e3-772d1b33fd14.png)
 
windows setting / region settings/

change the system locale to English.
![1](https://user-images.githubusercontent.com/47036391/153735169-6ba16c91-1100-4f97-b88b-86aea8c81a8a.png)

**Comment by mrei-kiel** [comment]
I got the same problem after updating to HDiNFRO64 v7.20-4700.
I haven't changed any reginal formatting, my region is set to German, though Language is English and the format is in English(German).
Previously everything was displaying fine in °C not in Â°C.  Entering this %3.0f°C helped.

**Comment by malikm** [comment]
HWiNFO64 since version 7.17 has been ported to UNICODE and as a result some of the text exported to other applications (like the Stream Deck plugin) might need a change to properly handle this. HWiNFO still exports the text in ANSI format, but now it's converted per the current system locale. So even for the English language the degree sign might be affected as some regional formats (e.g. Swedish, Danish, etc.) don't use the default ANSI character 0xB0, but the UTF-8 sequence 0xC2, 0xB0. When this is not properly handled (per current locale) by the 'consuming' application it will result in the "Â" displayed.

**Comment by malikm** [comment]
FYI, the Shared Memory object exposed by HWiNFO has been recently updated to export text items in UTF-8 format:
https://www.hwinfo.com/forum/threads/shared-memory-interface-v2-extension-to-support-utf-8-strings.8462/
This should allow applications like this plugin to properly obtain strings encoded in other locales.
Will require a small change in the plugin...

**Comment by blankster** [comment]
> This will show the number as ###°C %3.0f°C

Thank you, that workaround helped me a lot 🙏

**Comment by SRSLYRTFM** [comment]
> Hello, As long as you set HWInfo to Fahrenheit or Celcius, then you can use the formatter to display the number correctly. Then you can probably set you locale to what ever you want.
> 
> This will show the number as ###°F %3.0f°F
> 
> This will show the number as ###°C %3.0f°C
> 
> ![image](https://user-images.githubusercontent.com/99476709/153573840-17938645-e9a7-4f49-b8e3-772d1b33fd14.png)

Thank you so much for this workaround

**Comment by moeilijk** [comment]
This project is under new maintenance. The plugin core was replaced with a new engine in v3.0, now on main. A GitHub release follows shortly; tiles from v2.0.5 keep working after upgrading. If v3.0 does not do what you expected here, please open a new issue.

Fixed on main: shared memory strings are now decoded as UTF-8, so degree signs and other special characters render correctly (commit 46774b3).
